src/commands/createFunction.ts

```diff
@@ -25,6 +25,8 @@ import * as workspaceUtil from '../utils/workspace';
 import { VSCodeUI } from '../VSCodeUI';
 import { createNewProject } from './createNewProject';
 
+const functionNameRegex: RegExp = /^[a-zA-Z][a-zA-Z\d_\-]*$/;
```

Comment:
Don't use global variables when you don't have to. Put this in the context of `validateTemplateName`

(`requiredFunctionAppFiles` is a bad example because it was used in multiple places when the code was originally written)